app/classes/Transvision/Consistency.php

```diff
@@ -0,0 +1,121 @@
+<?php
+namespace Transvision;
+
+/**
+ * Consistency class
+ *
+ * Functions used to analyze consistency
+ *
+ * @package Transvision
+ */
+class Consistency
+{
+    /**
+     * Find duplicated strings (strings with the same text, ignoring case).
+     *
+     * @param array $strings_array Array of strings in the form
+     *                             string_id => string_value
+     *
+     * @return array Array of duplicated strings, same format as the
+     *               input array
+     */
+    public static function findDuplicates($strings_array)
+    {
+        $duplicates = [];
+        asort($strings_array);
+
+        $previous_key = '';
+        $previous_value = '';
+        foreach ($strings_array as $key => $value) {
+            if (strcasecmp($previous_value, $value) === 0) {
+                $duplicates[$previous_key] = $previous_value;
+                $duplicates[$key] = $strings_array[$key];
+            }
+            $previous_value = $value;
+            $previous_key = $key;
+        }
+
+        return $duplicates;
+    }
+
+     /**
+      * Filter out strings that should not be evaluted for consistency.
+      *
+      * @param  array $strings_array Array of strings in the form
+      *                              string_id => string_value
+      * @param  string Repository identifier
+      *
+      * @return array Array of filtered strings, with known false positives
+      *               removed
+      */
+     public static function filterStrings($strings_array, $repo)
+     {
+         if (in_array($repo, Project::getDesktopRepositories())) {
+             $repository_type = 'desktop';
+         } elseif (in_array($repo, Project::getGaiaRepositories())) {
+             $repository_type = 'gaia';
+         } else {
+             $repository_type = $repo;
+         }
+
+         $ignore_string = function ($key, $value) use ($repository_type) {
+             // Determine if a string should be excluded.
```

Comment:
put the comment before the anonymous function to make it readable with folding on